### PR TITLE
Lookup endpoint change

### PR DESCRIPTION
**Pull request by Joel-Joseph-George**
#676

### CODE REVIEW

File: app/crud/projects_crud.py [hunk 1]
```diff
@@ -24,7 +20,7 @@
 
 #pylint: disable=W0143,E1101
 ###################### Translation Project Mangement ######################
-def create_translation_project(db_:Session, project, user_id=None,app=None):
```
Comment (shimilgithub) [inline app/crud/projects_crud.py]: why reverting these changes?

File: app/crud/projects_crud.py [hunk 2]
```diff
@@ -37,8 +33,6 @@ def create_translation_project(db_:Session, project, user_id=None,app=None):
         meta['stopwords'] = project.stopwords.__dict__
     if project.punctuations:
         meta['punctuations'] = project.punctuations
-    if project.compatibleWith is None:
-        project.compatibleWith = [app]
```
Comment (shimilgithub) [inline app/crud/projects_crud.py]: it should be there

File: app/crud/projects_crud.py [hunk 3]
```diff
@@ -196,54 +187,10 @@ def update_translation_project(db_:Session, project_obj, project_id, user_id=Non
...
-                raise HTTPException(status_code=404,detail = \
-                    f"Project with id, {project_id}, not present")
-            if client_app not in compatible_with:
-                raise PermissionException("Incompatible app")
-        else:
-            raise PermissionException("Incompatible app")
-        return await func(*args, **kwargs)
-    return wrapper
-# pylint: enable=duplicate-code
-
```
Comment (shimilgithub) [inline app/crud/projects_crud.py]: these changes should not be reverted

File: app/crud/projects_crud.py (hunk 1 above)
Comment (Joel-Joseph-George) [reply app/crud/projects_crud.py]: i think happened when i merged the backup which reverted it back.

can you let me know which all files were changes reverted

**Review by shimilgithub — CHANGES_REQUESTED** [review]
All the changed related to `app_compatibility_check` is reverted back in this pr.  It has to be there.

File: app/crud/projects_crud.py (hunk 1 above)
Comment (shimilgithub) [reply app/crud/projects_crud.py]: you can refer this pr https://github.com/Bridgeconn/vachan-api/pull/665

File: app/routers/translation_apis.py [hunk 4]
```diff
@@ -736,6 +736,9 @@ async def generate_stopwords(request: Request, background_tasks: BackgroundTasks
     log.info('In generate_stopwords')
     log.debug('language_code:%s, use_server_data:%s, source_name:%s, sentence_list:%s',
         language_code, use_server_data, source_name, sentence_list)
+    print("language_Code,serverdata,sourcename,userdetails",
+          language_code,use_server_data,source_name,user_details)
+    print("sentencelist", sentence_list)
```
Comment (shimilgithub) [inline app/routers/translation_apis.py]: print statements can be removed

File: app/test/test_bible_books.py [hunk 5]
```diff
@@ -26,6 +26,7 @@ def test_get_book_code():
     '''positive test case, with one optional params, code'''
     #Without Auth
     response = client.get(UNIT_URL+'?book_code=psa')
+    print("####", response.json())
```
Comment (shimilgithub) [inline app/test/test_bible_books.py]: remove this

File: docker/Dockerfile [hunk 6]
```diff
@@ -1,4 +1,4 @@
-FROM tiangolo/uvicorn-gunicorn:python3.10
+FROM python:3.10
```
Comment (shimilgithub) [inline docker/Dockerfile]: it should not be reverted

File: docker/docker-compose.yml [hunk 7]
```diff
@@ -66,7 +66,7 @@ services:
       - VE-network
 
  vachan-db:
-    image: postgres:15.3
+    image: postgres:12.7
```
Comment (shimilgithub) [inline docker/docker-compose.yml]: we are using postgres 15.3 now. dont revert changes

**Review by shimilgithub — CHANGES_REQUESTED** [review]
Still some updates are reverted back in dockerFile and docker-compose file

File: app/test/test_stop_words_generation.py [hunk 8]
```diff
@@ -10,8 +10,8 @@
 from . test_auth_basic import login,SUPER_PASSWORD,SUPER_USER,logout_user
 
 
-UNIT_URL = '/v2/lookup/stopwords'
-GER_URL = '/v2/nlp/stopwords'
+UNIT_URL = '/v2/nlp/stopwords'  #lookup one
```
Comment (shimilgithub) [inline app/test/test_stop_words_generation.py]: When leaving comments, make it more clear so everyone can understand.

File: app/auth/authentication.py [hunk 9]
```diff
@@ -96,10 +96,22 @@ def api_resourcetype_map(endpoint, path_params=None):
...
+         endpoint.startswith("/v2/nlp"):
+        print("Endpoint", endpoint)
+        resource_type = schema_auth.ResourceType.TRANSLATION.value
+        #endpoint.split('/')[3] in ["stopwords/generate"]:
+        #endpoint.endswith(("stopwords/generate")):
+        # endpoint.split('/')[2] in ["generate"]:
+    #  or \
+    #      endpoint.endswith(("generate-stopwords"))
+    #  or (
+    #     path_params is not None and "language_Code" in path_params)
```
Comment (shimilgithub) [inline app/auth/authentication.py]: Eliminate any unused code that's been commented out by you to maintain a clean codebase.

File: app/routers/translation_apis.py [hunk 10]
```diff
@@ -67,7 +67,7 @@ async def create_project(request: Request,
 @get_auth_access_check_decorator
 @check_app_compatibility_decorator
 async def update_project(request: Request, project_obj:schemas_nlp.TranslationProjectEdit,
-    project_id:int=Query(...,example="1022004"),
+                         project_id:int=Query(...,example="1022004"),
```
Comment (shimilgithub) [inline app/routers/translation_apis.py]: keep proper alignment.Line 71 and 72 are also the part(parameters) of `update_project` function

File: app/test/test_stop_words_generation.py [hunk 11]
```diff
@@ -294,7 +294,8 @@ def test_generate_stopwords():
     add_tw_dict(dict_table_name)
     headers_auth['Authorization'] = "Bearer"+" "+initial_test_users['BcsDev']['token']
 
-    response = client.post(GER_URL+'/generate?language_code=hi',headers=headers_auth)
+    response = client.post(GER_URL+'?language_code=hi',headers=headers_auth)
+    print("###gen", response.json())
```
Comment (shimilgithub) [inline app/test/test_stop_words_generation.py]: still there are few print statements added in this pr, which are not required and may increase running time. those has to be removed.

**Review by shimilgithub — CHANGES_REQUESTED** [review]
Some minor codebase cleaning(only in the changes included in this pr) has to be done. Please ensure future PRs adhere to best practices as discussed in the review.

**Review by shimilgithub — APPROVED** [review]

**Pull request merged**